packages/integrations/tailwind/src/index.ts

```diff
@@ -14,13 +16,59 @@ function getDefaultTailwindConfig(srcUrl: URL) {
 	};
 }
 
-export default function (): AstroIntegration {
+async function getUserConfig(projectRoot: URL, configPath?: string) {
+	const resolvedProjectRoot = fileURLToPath(projectRoot);
+	let userConfigPath: string | undefined;
+
+	if (configPath) {
+		const configPathWithLeadingSlash = /^\.*\//.test(configPath) ? configPath : `./${configPath}`;
+		userConfigPath = fileURLToPath(new URL(configPathWithLeadingSlash, `file://${resolvedProjectRoot}/`));
+	}
+
+	return await load('tailwind', { mustExist: false, cwd: resolvedProjectRoot, filePath: userConfigPath });
+}
+
+type IntegrationConfig =
+	| {
+			config?: {
+				/**
+				 * Path to your tailwind config file
+				 * @default 'tailwind.config.js'
+				 */
+				path?: string;
+				/**
+				 * Apply Astro's default Tailwind config as a preset
+				 * This is recommended to enable Tailwind across all components and Astro files
+				 * @default true
+				 */
+				applyAstroPreset?: boolean;
+			};
+	  }
+	| undefined;
+
+export default function (integrationConfig: IntegrationConfig): AstroIntegration {
```

Comment:
nit: I would love to start the pattern of calling this argument `options`. We already have AstroConfig, ViteConfig... my brain is melting with all the different config references we juggle!

"Integration Options" is something that we can document, and then this argument name can match it.

Reply:
I agree with this! I would typically do something like `options: TailwindOptions`

Reply:
Used @natemoo-re's suggestion here!